src/NonceManager.sol

```diff
@@ -0,0 +1,23 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.23;
+
+import {INonceManager} from "./interfaces/INonceManager.sol";
+import {MinimalDelegationStorage, MinimalDelegationStorageLib} from "./libraries/MinimalDelegationStorage.sol";
+
+/// @title NonceManager
+/// @notice A contract that manages nonces to prevent replay attacks
+abstract contract NonceManager is INonceManager {
+    /// @inheritdoc INonceManager
+    function getNonce(uint192 key) public view returns (uint256 nonce) {
+        return MinimalDelegationStorageLib.get().nonceSequenceNumber[key] | (uint256(key) << 64);
+    }
+
+    /// @inheritdoc INonceManager
+    function validateAndUpdateNonce(uint256 nonce) public {
```

Comment:
this should be internal, otherwise anyone could invalidate your nonce!

also maybe name change?

_useNonce ?